src/test/librbd/test_librbd.cc

```diff
@@ -2332,6 +2354,160 @@ TEST_F(TestLibRBD, TestEncryptionLUKS2)
   rados_ioctx_destroy(ioctx);
 }
 
+#ifdef HAVE_LIBCRYPTSETUP
+
+TEST_F(TestLibRBD, TestCloneEncryption)
+{
+  REQUIRE(!is_feature_enabled(RBD_FEATURE_JOURNALING));
+  REQUIRE_FEATURE(RBD_FEATURE_LAYERING);
+
+  rados_ioctx_t ioctx;
+  rados_ioctx_create(_cluster, m_pool_name.c_str(), &ioctx);
+  ASSERT_EQ(0, rados_conf_set(
+          _cluster, "rbd_read_from_replica_policy", "balance"));
+
+  // create base image, write 'a's
+  int order = 12;
```

Comment:
This test writes at fixed offsets given the object size to craft specific copy-ups and layout.
i.e. write an object, clone, then write an object on the new child (which does not overlap with previous object), and so on.

Reply:
Those offsets appear to be 0, 64M and 128M.  Wouldn't 4M object size work just the same?

Reply:
Some tests turn on striping.
Perhaps the order can be increased, but not 4MB.

Reply:
Even with fancy striping in the mix, I'm not convinced.  The offsets are obviously picked on purpose but AFAICS you aren't asserting on specific copyups or layout.  All this test does is write to a bunch of image offsets and then read from them.

I might be missing something, but in such a scenario, striping is supposed to be completely transparent.  If it changes something with respect to this test, I'd like to know about it.  Can you paste the failure you ran into with 4M objects here?

Reply:
I'm just trying to craft the chain of clones so that the final read from the last clone (child3) actually yields reading from all ancestors, and not copy-upped data.

Reply:
... but you are not asserting on that because it is cumbersome, I see.  In that case, I would switch to default object size and disable this test case for striping with `is_feature_enabled`, just like you already do for journaling.